lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -4746,14 +4763,23 @@ LogicalResult FIRRTLLowering::visitPrintfLike(T op, StringAttr outputFile) {
             builder.getStringAttr("fd_" + outputFile.getValue()));
 
         addToInitialBlock([&]() {
-          auto constant = builder.create<sv::ConstantStrOp>(outputFile);
+          Value fileNameVal = builder.create<sv::ConstantStrOp>(outputFile);
+          if (outputFileAndRequiresSubst.getInt()) {
+            fileNameVal = builder
+                              .create<sv::SystemFunctionOp>(
+                                  hw::StringType::get(builder.getContext()),
+                                  builder.getStringAttr("sformatf"),
+                                  ValueRange{fileNameVal})
```

Comment:
Eventually we probably want to make this a first-class system function that CIRCT knows about.